### PR TITLE
Result

### PR DESCRIPTION
**Pull request by Anna710915**
clean up controllers, fix a validator and an update certificate service method, add new method 'deleteKeys' into the certificate repository

### CODE REVIEW

File: controller/src/main/java/com/epam/esm/controller/CertificateController.java [hunk 1]
```diff
@@ -153,11 +149,11 @@ public List<CertificateDto> searchCertificates(@RequestParam("part") String part
     public List<CertificateDto> sort(@RequestParam(value = "param_1", required = false) String name,
                                      @RequestParam(value = "param_2", required = false) String date){
         List<CertificateDto> certificateDtoList;
-        if(name.equals("name") && date.equals("date")){
+        if(name != null && date != null && name.equals("name") && date.equals("date")){
```
Comment (KseniyaYudzichavaTR) [inline controller/src/main/java/com/epam/esm/controller/CertificateController.java]: Awful statement - just write "name".equals(name) and so on

File: controller/src/main/java/com/epam/esm/controller/CertificateController.java [hunk 2]
```diff
@@ -153,11 +149,11 @@ public List<CertificateDto> searchCertificates(@RequestParam("part") String part
...
             certificateDtoList = certificateService.bothSort();
```
Comment (KseniyaYudzichavaTR) [inline controller/src/main/java/com/epam/esm/controller/CertificateController.java:155]: Do these methods sort only in ASC order?

File: persistence/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java [hunk 3]
```diff
@@ -81,6 +81,9 @@ INSERT INTO gift_certificate(name, description, price, duration,
     private static final String SQL_INSERT_GENERATED_KEYS = """
             INSERT INTO tags_gift_certificates(tag_id, gift_certificate_id)
             VALUES (?,?)""";
+    private static final String SQL_DELETE_KEYS = """
```
Comment (KseniyaYudzichavaTR) [inline persistence/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java:81]: Ops.... very strange - use CASCADE or what's the sense of this sql

File: service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java [hunk 4]
```diff
@@ -98,16 +98,18 @@ public  CertificateDto update(CertificateDto certificateDto, long id) {
         if(uniqCertificate != null && uniqCertificate.getId() != id){
             throw new IllegalArgumentException("Certificate name is not uniq " + certificateDto.getName());
         }
-        boolean modify = updateData(giftCertificate, certificateDto);
-        if(modify){
+        if(updateData(giftCertificate, certificateDto)){
```
Comment (KseniyaYudzichavaTR) [inline service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java]: public  CertificateDto update(CertificateDto certificateDto, long id) - logic of the method is incorrect

File: service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java [hunk 5]
```diff
@@ -121,10 +123,10 @@ public List<CertificateDto> showByTagName(String tagName) {
     public List<CertificateDto> showByPartWord(String partWord) {
```
Comment (KseniyaYudzichavaTR) [inline service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java:120]: search by part of name/description (can be implemented, using DB function call) - that means - if any of columns (name or description) contains a part of a word  - you are to select this GiftCertificate - incorrect current logic

File: controller/src/main/java/com/epam/esm/controller/CertificateController.java (hunk 2 above)
Comment (Anna710915) [reply controller/src/main/java/com/epam/esm/controller/CertificateController.java:155]: Yes, they do

File: service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java (hunk 5 above)
Comment (Anna710915) [reply service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java:120]: Okey, probably, I got you. Try to fix.

File: service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java (hunk 4 above)
Comment (Anna710915) [reply service/src/main/java/com/epam/esm/service/impl/CertificateServiceImpl.java]: Couldn't you explain more?

File: persistence/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java (hunk 3 above)
Comment (Anna710915) [reply persistence/src/main/java/com/epam/esm/repository/impl/CertificateRepositoryImpl.java:81]: Hmmm, I have cascade, but only on delete operations. And I know nothing about cascade insert.

File: controller/src/main/java/com/epam/esm/controller/CertificateController.java (hunk 1 above)
Comment (Anna710915) [reply controller/src/main/java/com/epam/esm/controller/CertificateController.java]: I got it, everything will be fixed

**Pull request merged**